src/main/java/com/fasterxml/jackson/databind/CacheProvider.java

```diff
@@ -0,0 +1,62 @@
+package com.fasterxml.jackson.databind;
+
+import com.fasterxml.jackson.databind.util.LookupCache;
+
+/**
+ * Container for {@link LookupCache} instances to use to override default cache implementations used.
+ * Should only be configured via {@link com.fasterxml.jackson.databind.json.JsonMapper.Builder#cacheProvider(CacheProvider)}.
+ * 
+ * @since 2.16
+ */
+public class CacheProvider
```

Comment:
I would prefer if we just had a cacheProvider with one create method.

`public LookupCache<K,V> createLookupCache(K k, V v)`

And other methods that to set the max size on each type of cache. One example:

`public void setSerializerCacheMaxSize(int size)`

These set methods would only affect caches created after the set was called. We would not attempt to replace the existing caches when the setSerializerCacheMaxSize is called.

Reply:
I disagree: different use cases may want to use different implementations, so separate calls make sense to me.

Similarly size-setting etc cannot really go through plain `CacheProvider` unless we hard-code construction, or require `LookupCache` to have set method (and be mutable).

But I think what would work to achieve same end, is to provide one default implementation that has such "simple" set methods and constructs default `LookupCache` instances with given sizes.
This provider could also be used as a base class for other somewhat simple implementations.

Reply:
Could we at least consider having an Interface so that users can provide their own implementations? We can have this implementation class extend the interface

Reply:
Makes sense @pjfanning